### DISCUSSION TITLE
Preflight reset as a class

### DISCUSSION THREAD

**Discussion by Soviut** [tailwindlabs/tailwindcss#7710] [Ideas]
I've integrated Tailwind into a couple existing sites and while disabling preflight is easy enough but having to deal with inherited site styles can be annoying.

I think it would be a good idea to make it possible to configure the preflight reset as a class, similar to dark mode.

```javascript
{
  preflight: 'class'
}
```

Then, a class (either `.preflight` or `.reset`) could be used to surround regions you'd like to use tailwind classes without inheriting anything from the rest of the site.

```html
<div class="reset">
  <div class="p-5 border rounded">
    <p class="mb-3">Do you agree to these terms?</p>
    <button class="p-3 rounded bg-teal-500 text-white">Agree</button>
  </div>
</div>
```

There are probably some issues with specificity that I'm not considering, but the overall implementation should be fairly sound.

**Comment by k-utsumi** [comment]
Probably the same or close to https://github.com/tailwindlabs/tailwindcss/discussions/7050 (Comment for link)
